src/ApiCore/Middleware/OperationsCallable.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/*
+ * Copyright 2018, Google Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+namespace Google\ApiCore\Middleware;
+
+use Google\ApiCore\Call;
+use Google\ApiCore\LongRunning\OperationsClient;
+use Google\ApiCore\OperationResponse;
+use Google\Protobuf\Internal\Message;
+
+/**
+ * Callable which wraps the response in an OperationResponse object.
+ */
+class OperationsCallable
+{
+    /** @var callable */
+    private $nextHandler;
+
+    /** @var OperationsClient */
+    private $client;
+
+    /** @var array */
+    private $descriptor;
+
+    public function __construct(
+        callable $nextHandler,
+        OperationsClient $client,
+        array $descriptor
+    ) {
+        $this->nextHandler = $nextHandler;
+        $this->client = $client;
+        $this->descriptor = $descriptor;
+    }
+
+    public function __invoke(Call $call, array $options)
+    {
+        $next = $this->nextHandler;
+        $client = $this->client;
```

Comment:
There is an issue where with a callable you can't instantiate it from class scope, e.g.:

```php
$this->nextHandler($call, $options); // blows up
```

But for the other class level variables (client, descriptor), we can access those without the need for the use statement using `$this`. I think this is preferable and a bit cleaner.

Reply:
Yes, it definitely works. It was a feature added in PHP 5.4.

http://php.net/manual/en/functions.anonymous.php#example-167

I wonder if your IDE is using rules defined for PHP 5.3 or lower?